autometa/taxonomy/ncbi.py

```
@@ -364,60 +345,64 @@ def parse_merged(self):
-        description='Autometa NCBI utilities class')
-    parser.add_argument('ncbi', help='</path/to/ncbi/database/directory>')
-    parser.add_argument(
-        '--query-taxid',
-        help='query taxid to test NCBI class functionality',
-        default=1222,
-        type=int,
-    )
-    parser.add_argument('--verbose', help="add verbosity",
-                        action='store_true', default=True)
-    args = parser.parse_args()
-    ncbi = NCBI(args.ncbi, args.verbose)
-    query_taxid_parent = ncbi.parent(taxid=args.query_taxid)
-    logger.info(
-        f'{args.query_taxid} Name: {ncbi.name(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} Order: {ncbi.name(taxid=args.query_taxid, rank="order")}\n'
-        f'{args.query_taxid} Class: {ncbi.name(taxid=args.query_taxid, rank="class")}\n'
-        f'{args.query_taxid} Rank: {ncbi.rank(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} Lineage:\n{ncbi.lineage(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} is_common_ancestor {query_taxid_parent}: '
-        f'{ncbi.is_common_ancestor(parent_taxid=query_taxid_parent, child_taxid=args.query_taxid)}\n'
-    )
+        taxid_A_lineage_taxids = {taxids.get('taxid')
+                                  for taxids in self.lineage(taxid_A)}
+        taxid_B_lineage_taxids = {taxids.get('taxid')
+                                  for taxids in self.lineage(taxid_B)}
+        common_ancestor = taxid_B_lineage_taxids.intersection(
+            taxid_A_lineage_taxids)
+        common_ancestor.discard(1)  # This discards root
+        return True if common_ancestor else False
+
+    def is_valid_taxid(self, taxid):
+        """
+        Checks if the given taxid is a positive interger or not 
+
+        Parameters
+        ----------
+        `taxid` : int
+            identifer for a taxon in NCI taxonomy databses - `nodes.dmp`, `names.dmp` or `merged.dmp`
+
+        Returns
+        -------
+        `taxid` or boolean
```

Instead of returning an `int` and a `boolean`, why not return an `int` that does not exist in the database that we know is invalid? Like `-1` or `0`?

If we were to use `0`, this would also evaluate to false and we would have only one return type.

---

Actually, looking through the rest of the script. When you are applying `self.is_valid_taxid`, every time the returned value is `False`, you raise a `ValueError`. You might as well raise the `ValueError` in the `is_valid_taxid` method. If the error is _not_ encountered, then return the converted value. This should still behave similar to my comments from Lines 389-402 but you should not need to convert an invalid taxid to `-1` or `0`, you can just raise the error as you are doing when you are calling the method `self.is_valid_taxid`. Then the exception handling in the respective methods (`self.name`, `self.rank`, `self.parent`) can be removed.